src/custom/connectors/index.ts

```diff
@@ -1,16 +1,110 @@
-import { ChainId } from '@uniswap/sdk'
+import { Web3Provider } from '@ethersproject/providers'
 import { InjectedConnector } from '@web3-react/injected-connector'
+import { WalletConnectConnector } from '@web3-react/walletconnect-connector'
+import { WalletLinkConnector } from '@web3-react/walletlink-connector'
+import { PortisConnector } from '@web3-react/portis-connector'
 
-export {
-  NETWORK_CHAIN_ID,
-  fortmatic,
-  portis,
-  network,
-  walletconnect,
-  walletlink,
-  getNetworkLibrary
-} from '@src/connectors'
-
-export const injected = new InjectedConnector({
-  supportedChainIds: [ChainId.MAINNET, ChainId.ROPSTEN, ChainId.RINKEBY, ChainId.GÖRLI, ChainId.KOVAN, ChainId.XDAI]
+import { FortmaticConnector } from 'connectors/Fortmatic'
+import { NetworkConnector } from 'connectors/NetworkConnector'
+
+const SUPPORTED_CHAIN_IDS = process.env.REACT_APP_SUPPORTED_CHAIN_IDS
+const DEFAULT_CHAIN_ID = Number(process.env.REACT_APP_CHAIN_ID || '1')
+
+const FORMATIC_KEY = process.env.REACT_APP_FORTMATIC_KEY
+const PORTIS_ID = process.env.REACT_APP_PORTIS_ID
+
+export const NETWORK_CHAIN_ID: number = parseInt(process.env.REACT_APP_CHAIN_ID ?? '1')
+
+type RpcNetworks = { [chainId: number]: string }
+
+function getRpcNetworks(): [RpcNetworks, number[]] {
+  // Make sure the mandatory envs are present
+  if (typeof SUPPORTED_CHAIN_IDS === 'undefined') {
+    throw new Error(`REACT_APP_NETWORK_URL must be a defined environment variable`)
+  }
+
+  // Get list of supported chains
+  const chainIds = SUPPORTED_CHAIN_IDS.split(',').map(chainId => Number(chainId.trim()))
```

Comment:
can you make this `split` outside the function body? or even the entire constant assignment

Reply:
i moved them in

Reply:
@anxolin well my comment was the opposite.

im assuming this function is only called once on load, in which case it's fine. but if we expect it to be called repeatedly then it could be an issue to always be splitting the networks array env + mapping etc since it won't change between loads

Reply:
this is an initialization, only runs once. This is why I thought there's no reason to keep some vars in scope. When the function is over, we don't need any more the strings